main_test.go

```diff
@@ -713,3 +716,37 @@ func Test_runScorecardSettings(t *testing.T) {
 		})
 	}
 }
+
+func Test_signScorecardResult(t *testing.T) {
+	t.Parallel()
+	// Sign example scorecard results file.
+	scorecardResultsFile := "./testdata/scorecard-results-example.sarif"
+	err := signScorecardResult(scorecardResultsFile)
+	if err != nil {
+		t.Errorf("signScorecardResult() error, %v", err)
+		return
+	}
+
+	// Verify that the signature was created and uploaded to the Rekor tlog by looking up the payload.
+	ctx := context.Background()
+	rekorClient, err := rekor.NewClient("https://rekor.sigstore.dev")
+	if err != nil {
+		t.Errorf("signScorecardResult() error getting Rekor client, %v", err)
+		return
+	}
+	scorecardResultData, err := ioutil.ReadFile("./testdata/scorecard-results-example.sarif")
+	if err != nil {
+		t.Errorf("signScorecardResult() error reading scorecard result file, %v", err)
+		return
+	}
+	uuids, err := cosign.FindTLogEntriesByPayload(ctx, rekorClient, scorecardResultData)
+	if err != nil {
+		t.Errorf("signScorecardResult() error getting tlog entries, %v", err)
+		return
+	}
+
+	if len(uuids) == 0 {
```

Comment:
is this uuid known ahead of time? I mean, can we verify it matches the one returned by ... `signScorecardResult()`?
Not sure this is do-able, just asking in case

Reply:
> do we need to verify that the signature is valid?

One way to verify is by providing the certificate, signature, and transparency log index. @azeemshaikh38 mentioned that the GitHub Action will make a POST HTTP call to an endpoint that will process this verification, and the POST data payload will only contain the commitSHA, timestamp, and scorecard result payload. 

So instead of verifying normally we will use the scorecard payload to lookup all associated tlog entries, and find the one that matches our commitSHA. This way out endpoint can prove, given only a repo commit and a "claimed" scorecard result, that the results were actually generated from a scorecard run on the specified repo. 

The test in main_test.go currently doesn't account for signatures on different commits. It just submits a payload and makes sure its signature is being logged in Rekor. I wrote a function [here](https://github.com/rohankh532/scorecard-cosign-server/blob/8ced730728fdd98cf62b551e79b6cdc2acd0d68c/main.go#L32) that expands on this, and I'm working on validating the actual workflow yml contents too.

I think this design choice is so that repo owners can prove that their displayed scorecard results are authentic. Please correct if there are any misunderstandings.

Reply:
you understand correctly. If the verifying tool lives in this repo, we can perform the verification in our unit tests as well.
I agree this can be done in a follow-up PR. I was mostly asking to be sure the unit tests are complete.

@azeemshaikh38 @rohankh532 where do you think the verification code should live? In this repo or in the wepbapp repo? Im tempted to say in this repo, since both are tightly related... but I'm not entirely sure. Wdut?

Reply:
I think a standalone endpoint would be useful later on, like for badging and displaying results on scorecards.dev. So maybe the webapp repo?

Reply:
I think that'd work, yes. Let's see what @azeemshaikh38 thinks

Reply:
+1 to keeping the verification in the webapp repo. The webapp repo will be its own standalone API (GH Action being *one* of its clients), so keeping the verification there makes sense to me.